### PR TITLE
Improvements to plugin handler

### PR DESCRIPTION
**Pull request by ryanblock**
Changes deploy step order to put plugins after hydration

Thank you for helping out! ✨

Before submitting a pull request, please make sure the you checked the following:

- [x] Forked the repository and created your branch from master
- [x] Make sure the tests pass! `npm it` in the repository root
- [x] If you've fixed a bug or added code **more** tests are appreciated
- [x] If you're adding a new command, removing an old command, or changing how a command works, please update the relevant documentation for the command (under the `doc/` folder)
- [x] Summarize your changes in the [changelog](https://github.com/arc-repos/architect/blob/master/changelog.md)
- [x] If you haven't already please complete the CLA

Learn more about contributing: https://arc.codes/intro/community

### CODE REVIEW

File: src/util/run-plugin-promise.js [hunk 1]
```diff
@@ -21,16 +21,20 @@ module.exports = function runPluginFunction(params, functionName) {
...
         }
-
-        var tmp = require(pluginName)
-        var has = tmp.hasOwnProperty(functionName)
-        var fn = tmp[functionName]
+        if (pluginName.split('').includes('/')) {
+          pluginName = '@' + pluginName
+        }
+        // Must require from full path from via current Architect process
+        var req = process.cwd() + '/node_modules/' + pluginName
```
Comment (austinkelleher) [inline src/util/run-plugin-promise.js]: I think `path.join(...)` would be better to support multiple platforms (e.g. Windows)

File: src/util/run-plugin-promise.js [hunk 2]
```diff
@@ -21,16 +21,20 @@ module.exports = function runPluginFunction(params, functionName) {
         } else {
           pluginName = plugin
         }
-
-        var tmp = require(pluginName)
-        var has = tmp.hasOwnProperty(functionName)
-        var fn = tmp[functionName]
+        if (pluginName.split('').includes('/')) {
```
Comment (austinkelleher) [inline src/util/run-plugin-promise.js]: We should probably also check for a backslash for other platforms

File: src/util/run-plugin-promise.js (hunk 1 above)
Comment (ryanblock) [reply src/util/run-plugin-promise.js]: Hmm, excellent catch, require additional interpolation in there!

File: src/util/run-plugin-promise.js (hunk 2 above)
Comment (ryanblock) [reply src/util/run-plugin-promise.js]: This bit is there specifically for NPM scoping (eg `@architect/functions`) so we can rely on it being a forward slash, but per your point above it does need to be interpolated to backslash on Windows when running a local absolute path based `require`.

Comment (austinkelleher) [reply src/util/run-plugin-promise.js]: Ah got it. Thanks for explaining.

**Pull request merged**